logstash-core/lib/logstash/plugin.rb

```diff
@@ -46,8 +45,8 @@ def eql?(other)
   end
 
   def initialize(params=nil)
+    @logger = self.class.logger
```

Comment:
I took some time to look it into more details; Its is possible to a mixin to provide the instance variable but its really an hacky solution which requires doing an `alias_method` on the initialize. Lets no go that route @talevy and lets just keep the instance variable for now.